app/models/relationship.rb

```diff
@@ -0,0 +1,6 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: "User"
+  belongs_to :followed, class_name: "User"
```

Comment:
Rails 5+ allows you (and I think encourages you) to condense the relationship and the validation into a single line.

So this would become:

```suggestion
  belongs_to :followed, class_name: User.name, required: true
```

And then you can ditch the `validates :followed_id, ...` line